config/main.py

```diff
@@ -3369,37 +3369,67 @@ def get_acl_bound_ports():
 
     return list(ports)
 
-#
-# 'table' subcommand ('config acl add table ...')
-#
 
-@add.command()
-@click.argument("table_name", metavar="<table_name>")
-@click.argument("table_type", metavar="<table_type>")
-@click.option("-d", "--description")
-@click.option("-p", "--ports")
-@click.option("-s", "--stage", type=click.Choice(["ingress", "egress"]), default="ingress")
-def table(table_name, table_type, description, ports, stage):
+def expand_vlan_ports(port_name):
     """
-    Add ACL table
+    Expands a given VLAN interface into its member ports.
+
+    If the provided interface is a VLAN, then this method will return its member ports.
+
+    If the provided interface is not a VLAN, then this method will return a list with only
+    the provided interface in it.
     """
     config_db = ConfigDBConnector()
     config_db.connect()
 
+    vlan_members = config_db.get_keys("VLAN_MEMBER")
+
+    members = [v[1] for v in vlan_members if port_name == v[0]]
+    return members or [port_name]
+
+
+def parse_acl_table_info(table_name, table_type, description, ports, stage, expand_vlans):
     table_info = {"type": table_type}
 
     if description:
         table_info["policy_desc"] = description
     else:
         table_info["policy_desc"] = table_name
 
-    if ports:
+    if ports and expand_vlans:
+        port_list = list()
+        for port in ports.split(","):
+            port_list += expand_vlan_ports(port)
+        table_info["ports@"] = ",".join(set(port_list))
+    elif ports:
         table_info["ports@"] = ports
     else:
         table_info["ports@"] = ",".join(get_acl_bound_ports())
```

Comment:
>",".join(get_acl_bound_ports()) [](start = 31, length = 31)

ConfigDBConector could hanlde the translation between list to string. Maybe it's easier to reuse its functions.

Reply:
I'm not sure if we can since we have to expand each interface and then de-duplicate the results and put them together.